services/web/server/src/simcore_service_webserver/rest_healthcheck.py

```diff
@@ -0,0 +1,93 @@
+""" Service healthcheck
+
+From https://docs.docker.com/engine/reference/builder/#healthcheck
+
+    --interval=DURATION (default: 30s)
+    --timeout=DURATION (default: 30s)
+    --start-period=DURATION (default: 0s)
+    --retries=N (default: 3)
+
+    The health check will first run *interval* seconds after the container is started, and
+    then again *interval* seconds after each previous check completes.
+
+    If a single run of the check takes longer than *timeout* seconds then the check is considered to have failed (SEE HealthCheckFailed).
+
+    It takes *retries* consecutive failures of the health check for the container to be considered **unhealthy**.
+
+    *start period* provides initialization time for containers that need time to bootstrap. Probe failure during
+    that period will not be counted towards the maximum number of retries.
+
+    However, if a health check succeeds during the *start period*, the container is considered started and all consecutive
+    failures will be counted towards the maximum number of retries.
+
+"""
+
+
+import asyncio
+import inspect
+from typing import TYPE_CHECKING, Any, Awaitable, Callable, Dict, Optional
+
+from aiohttp import web
+from aiosignal import Signal
+
+from ._constants import APP_SETTINGS_KEY
+
+if TYPE_CHECKING:  # pragma: no cover
+    _HeathCheckSignal = Signal[Callable[[web.Application], Awaitable[None]]]
+
+else:
+    _HeathCheckSignal = Signal
+
+
+class HealthCheckFailed(RuntimeError):
+    """Failed a health check
+
+    NOTE: not the same as unhealthy. Check module's doc
+    """
+
+
+class HeathCheck:
+    def __init__(self):
+        self._on_healthcheck = Signal(owner=self)  # type: _HeathCheckSignal
+
+    @property
+    def on_healthcheck(self) -> _HeathCheckSignal:
+        return self._on_healthcheck
+
+    @staticmethod
+    def get_app_info(app: web.Application):
```

Comment:
this can be cached

Reply:
it is not necessary. It is a dict with constants. The app-settings are basically constants.